### ISSUE TITLE
Your standard is impractical

### ISSUE THREAD

**Issue by partycoder** [feross/standard#174]
It does not mention anything about:
* Variadic functions
* Function length
* Statement length
* Function cyclomatic complexity
* Statement cyclomatic complexity
* Redefining variables
* Setting values to function arguments
* Dealing with numbers: exact comparisons with floating point numbers, checking for NaN/Infinity
* Dealing with dates
* Dealing with booleans (e.g: name all booleans, use a prefix like "can", "has", "is")
* .... (all the things that make a standard practical).

In short, people can respect your standard and still produce horrible code.
You have emphasized whitespace but not true *readability*.
Therefore, fix your standard.

**Comment by jprichardson** [comment]
`standard` doesn't prevent people from writing horrible code as "horrible" is not an objective measure but a subjective one. `standard` sets a baseline, removes thought from silly decisions, and increases team cohesion. `standard` is a start, not the panacea. 

Fortunately `standard` is extremely easy to extend; you can extend it with rules that satisfy your stated requirements. I'd suggest that you go grab [npm:standard-nonhorrible](https://www.npmjs.com/package/standard-nonhorrible) as it's still available :p

**Comment by partycoder** [comment]
Some items I mentioned here are quantifiable metrics.
And "horrible code" can be subjective, but the criteria I mentioned is not subjective.

 `standard` achieves not much in terms of quality, you should remove this claim from your README.md file:
```
This module helps hold our code to a high standard of quality.
```

That is only possible if the quality of your code was already lower than what this standard allows, which is very unlikely.

**Comment by zanona** [comment]
@arboreal84, by any chance would you be able to share the ESLint rules for your suggestion?
@jprichardson how would you recommend extending these rules into `standard`?
Thanks in advance

**Comment by arboreal84** [comment]
(Well first of all, sorry for the initial tone).

- maximum statements per function
https://eslint.org/docs/rules/max-statements

- maximum cyclomatic complexity per function
https://eslint.org/docs/rules/complexity

- assignments to function parameters
https://eslint.org/docs/rules/no-param-reassign

- statement length (through line length)
https://eslint.org/docs/rules/max-len

Now there's also "prefer constants", which will let you know if a variable can be turned into a `const`.
More enforcement than this can be hard to achieve in JS.
https://eslint.org/docs/rules/prefer-const

To my knowledge there's no static analysis for numbers, dates, etc.

**Comment by zanona** [comment]
Thanks a lot for sharing those @arboreal84, it's super helpful.
Now it would be great to hear from @jprichardson what's the recommended way to extend these into `standard`.

The only way I can think of is to have a `.eslintrc` file only with these declarations and use both `eslint` and `standard` to test your code. but then, wouldn't it be better to use https://github.com/standard/eslint-config-standard altogether instead?

**Comment by arboreal84** [comment]
Well, as @jprichardson mentioned, the objective of `standard` is to create some sort of least common denominator that appeals to most people (i.e: maximize adoption), rather than creating a bulletproof standard.

That's why I suggested establishing levels,
e.g:
- standard level 0 = what you have today
- standard level 1 = additional rules with relaxed defaults
- standard level 2 = more rules, more strict defaults

There's a lot of room for improvement with it comes to coding standards and static analysis.

**Comment by feross** [comment]
@zanona You may wish to look into [`standardx`](https://github.com/standard/standardx) which allows you to layer in additional custom rules.
